lib/rucio/daemons/bb8/common.py

```diff
@@ -220,36 +225,52 @@ def list_rebalance_rule_candidates(rse, mode=None, session=None):
     :param session:      DB Session.
     """
 
-    if mode is None:
-        sql = """SELECT dsl.scope as scope, dsl.name as name, rawtohex(r.id) as rule_id, r.rse_expression as rse_expression, r.subscription_id as subscription_id, d.bytes as bytes, d.length as length, (case when d.length < 1 then 0 else d.bytes/d.length end) as fsize FROM atlas_rucio.dataset_locks dsl JOIN atlas_rucio.rules r ON (dsl.rule_id = r.id) JOIN atlas_rucio.dids d ON (dsl.scope = d.scope and dsl.name = d.name)
-WHERE
-dsl.rse_id = atlas_rucio.rse2id(:rse) and
-(r.expires_at > sysdate+60 or r.expires_at is NULL) and
-r.created_at < sysdate-60 and
-r.account IN ('panda', 'root', 'ddmadmin') and
-r.state = 'O' and
-r.copies = 1 and
-r.did_type = 'D' and
-r.child_rule_id is NULL and
-d.bytes is not NULL and
-d.is_open = 0 and
-d.did_type = 'D' and
-r.grouping IN ('D', 'A') and
-(case when d.length < 1 then 0 else d.bytes/d.length end) > 1000*1000*1000 and
-1 = (SELECT count(*) FROM atlas_rucio.dataset_locks WHERE scope=dsl.scope and name=dsl.name and rse_id = dsl.rse_id)
-ORDER BY fsize DESC, dsl.accessed_at ASC NULLS FIRST, d.bytes DESC"""  # NOQA
-    elif mode == 'decommission':  # OBSOLETE
-        sql = """SELECT r.scope, r.name, rawtohex(r.id) as rule_id, r.rse_expression as rse_expression, r.subscription_id as subscription_id, 0 as bytes, 0 as length FROM atlas_rucio.rules r
-WHERE
-r.id IN (SELECT rule_id FROM atlas_rucio.locks WHERE rse_id = atlas_rucio.rse2id(:rse) GROUP BY rule_id) and
-r.state = 'O' and
-r.child_rule_id is NULL"""  # NOQA
-
-    if mode != 'decommission':
-        return session.execute(sql, {'rse': rse}).fetchall()
-    else:  # can be applied only for decommission since the dumps doesn't contain info from dids and rules tables.
+    rse_id = get_rse_id(rse)
+
+    # dumps can be applied only for decommission since the dumps doesn't contain info from dids
+    if mode == 'decommission':
         return _list_rebalance_rule_candidates_dump(rse, mode)
 
+    # the rest is done with sql query
+    from_date = datetime.utcnow() + timedelta(days=60)
+    to_date = datetime.now() - timedelta(days=60)
+    allowed_accounts = ['panda', 'root', 'ddmadmin']
```

Comment:
That sqlalchemy query is a really nice job 😄 
In the long-term, there is a few things in there which we should maybe make configurable, so this can be used by other experiments, e.g. allowed_accounts, but for now it looks really good like this.